src/dmd/backend/cg87.d

```diff
@@ -646,53 +644,45 @@ static void genftst(CodeBuilder& cdb,elem *e,int pop)
  *      0 if not
  */
 
-unsigned char loadconst(elem *e, int im)
-#if __DMC__
-__in
+ubyte loadconst(elem *e, int im)
 {
     elem_debug(e);
     assert(im == 0 || im == 1);
-}
-__body
-#endif
-{
-    static float fval[7] =
-        {0.0,1.0,PI,LOG2T,LOG2E,LOG2,LN2};
-    static double dval[7] =
-        {0.0,1.0,PI,LOG2T,LOG2E,LOG2,LN2};
-    static longdouble ldval[7] =
-#if __DMC__    // from math.h
-    {0.0,1.0,M_PI_L,M_LOG2T_L,M_LOG2E_L,M_LOG2_L,M_LN2_L};
-#elif _MSC_VER // struct longdouble constants
-    {ld_zero, ld_one, ld_pi, ld_log2t, ld_log2e, ld_log2, ld_ln2};
-#else          // C99 hexadecimal floats (GCC, CLANG, ...)
-#define M_PI_L          0x1.921fb54442d1846ap+1L        // 3.14159 fldpi
-#define M_LOG2T_L       0x1.a934f0979a3715fcp+1L        // 3.32193 fldl2t
-#define M_LOG2E_L       0x1.71547652b82fe178p+0L        // 1.4427 fldl2e
-#define M_LOG2_L        0x1.34413509f79fef32p-2L        // 0.30103 fldlg2
-#define M_LN2_L         0x1.62e42fefa39ef358p-1L        // 0.693147 fldln2
-    {0.0,1.0,M_PI_L,M_LOG2T_L,M_LOG2E_L,M_LOG2_L,M_LN2_L};
-#endif
-    static unsigned char opcode[7 + 1] =
+
+    immutable float[7] fval =
+        [0.0,1.0,PI,LOG2T,LOG2E,LOG2,LN2];
+    immutable double[7] dval =
+        [0.0,1.0,PI,LOG2T,LOG2E,LOG2,LN2];
+
+    enum M_PI_L        = 0x1.921fb54442d1846ap+1L;       // 3.14159 fldpi
+    enum M_LOG2T_L     = 0x1.a934f0979a3715fcp+1L;       // 3.32193 fldl2t
+    enum M_LOG2E_L     = 0x1.71547652b82fe178p+0L;       // 1.4427 fldl2e
+    enum M_LOG2_L      = 0x1.34413509f79fef32p-2L;       // 0.30103 fldlg2
+    enum M_LN2_L       = 0x1.62e42fefa39ef358p-1L;       // 0.693147 fldln2
+    immutable real[7] ldval =
```

Comment:
You should not use `real` directly, these are just 64-bit with LDC on Windows. Use `targ_ldouble` instead.